cli/src/oxide_override.rs

```diff
@@ -0,0 +1,600 @@
+                    .body_map(|body| body.allowed_ips(AllowedSourceIps::Any));
+            }
+            Some("ips") => {
+                let values: Vec<IpOrNet> = matches.get_many("ips").unwrap().cloned().collect();
+                *request = request.to_owned().body_map(|body| {
+                    body.allowed_ips(AllowedSourceIps::List(
+                        values.into_iter().map(IpOrNet::into_ip_net).collect(),
+                    ))
+                });
+            }
+            _ => unreachable!("invalid value for allow-list group"),
+        }
+
+        Ok(())
+    }
+}
+
+impl OxideOverride {
+    /// Construct a new OxideOverride for JSON output.
+    pub fn new_json() -> Self {
+        OxideOverride::Json {
+            needs_comma: AtomicBool::new(false),
+        }
+    }
+
+    /// Construct a new OxideOverride for tabular output.
+    pub fn new_table(fields: &[String]) -> Self {
+        OxideOverride::Table {
+            table: Box::new(Mutex::new(TableFormatter::new(fields))),
+        }
+    }
+
+    fn ip_range(matches: &clap::ArgMatches) -> anyhow::Result<IpRange> {
+        let first = matches.get_one::<IpAddr>("first").unwrap();
+        let last = matches.get_one::<IpAddr>("last").unwrap();
+
+        match (first, last) {
+            (IpAddr::V4(first), IpAddr::V4(last)) => {
+                let range = Ipv4Range::try_from(Ipv4Range::builder().first(*first).last(*last))?;
+                Ok(range.into())
+            }
+            (IpAddr::V6(first), IpAddr::V6(last)) => {
+                let range = Ipv6Range::try_from(Ipv6Range::builder().first(*first).last(*last))?;
+                Ok(range.into())
+            }
+            _ => anyhow::bail!(
+                "first and last must either both be ipv4 or ipv6 addresses".to_string()
+            ),
+        }
+    }
+}
+
+enum ReturnType {
+    Array,
+    Object,
+}
+
+/// Find the fields present on a the object returned by `success_item`.
+fn success_item_fields(root_schema: &RootSchema) -> (Vec<String>, ReturnType) {
+    let props = match (
+        root_schema.schema.object.as_ref().map(|o| &o.properties),
+        root_schema
+            .schema
+            .subschemas
+            .as_ref()
+            .and_then(|s| s.one_of.as_ref()),
+    ) {
+        // Generally the array items are a single type.
+        (Some(props), _) => props,
+        // Some endpoints return an enum, e.g.
+        // `/v1/anti-affinity-groups/{anti_affinity_group}/members/instance`.
+        // Collect the fields from all of the variants.
+        (_, Some(variants)) => {
+            return (collect_variant_fields(variants), ReturnType::Object);
+        }
+        // Some endpoints, e.g. `/v1/system/hardware/switch-port/{port}/status` will
+        // return an untyped JSON value. Just bail out when we hit these.
+        _ => {
+            return (Vec::new(), ReturnType::Object);
+        }
+    };
+
+    // Check if we're receiving an array.
```

Comment:
this seems to do the opposite i.e. it checks if we're **not** receiving an array.